src/SavingsDai.sol

```diff
@@ -76,6 +76,7 @@ contract SavingsDai {
     event Transfer(address indexed from, address indexed to, uint256 value);
     event Deposit(address indexed sender, address indexed owner, uint256 assets, uint256 shares);
     event Withdraw(address indexed sender, address indexed receiver, address indexed owner, uint256 assets, uint256 shares);
+    event ReferredDeposit(uint16 indexed referralCode, address indexed owner, uint256 assets, uint256 shares);
```

Comment:
Some comments for consideration:
1. Not 100% sure we want to re-emit the other params apart from `referralCode`, as it's fairly easy to fetch them from the previous event log in the tx.
2. Consider just calling the event `Refferal`, as `Deposit` and `ReferredDeposit` may confuse to think there are 2 deposits done. Also, as the event is is emitted also on `mint()` and not only on `deposit`, it avoids that confusion.
3. Consider renaming `referralCode` to just `referral` everywhere. I know in Aave it's `referralCode`, but the short pattern seems more fitting for the naming conventions here.

Reply:
1. Though it is indeed not strictly necessary, there is a case to be made for the convenience of getting all the relevant params in the same Referral event (thereby reducing the likelihood of an incorrect off-chain calculation). This does cost a bit of extra gas though so I'm open to both approaches.
2. Done
3. Done

Reply:
ok